### PR TITLE
Unify config api

### LINKED ISSUE #996: For SpanProcessor/SpanExporter/MetricProducer/MetricExporter use Config
This is a proposal to use a Config object instead of Builders for all the implementations of these classes. This will allow us to have a "getFromEnv" on all these Configs, probably a "getDefault" as well.

This will be easier to initialize compared with Builders in auto-instrumentation as well as when dealing with initializing a pipeline like SpanProcessor -> SpanExporter.

/cc @trask @tylerbenson - what do you think about this from the auto-instrumentation perspective.

### PR DESCRIPTION
**Pull request by thisthat**
This resolves #996

### CODE REVIEW

File: sdk/src/test/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessorTest.java [hunk 1]
```diff
@@ -329,14 +126,14 @@ public void exportDifferentSampledSpans() {
   @Test
   public void exportMoreSpansThanTheBufferSize() {
     WaitingSpanExporter waitingSpanExporter = new WaitingSpanExporter(6);
-    BatchSpansProcessor.Config config =
-        BatchSpansProcessor.Config.newBuilder()
+    BatchSpansProcessor bsp =
```
Comment (jkwatson) [inline sdk/src/test/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessorTest.java:129]: would prefer not abbreviating like this. We should prefer spelling out names to make reading the call sites less confusing.

Comment (bogdandrutu) [reply sdk/src/test/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessorTest.java:129]: To rephrase what @jkwatson said: we are Java not Go :)))

Comment (thisthat) [reply sdk/src/test/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessorTest.java]: Addressed in 0af5490

**Comment by bogdandrutu** [comment]
Please rebase

**Comment by thisthat** [comment]
> Please rebase

Done!

**Pull request merged**